src/sentry/migrations/0652_alert_rule_activation_condition.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 5.0.2 on 2024-02-22 18:50
+
+import django.db.models.deletion
+import django.utils.timezone
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0651_enable_activated_alert_rules"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="AlertRuleActivationCondition",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("label", models.TextField()),
+                ("condition_type", models.SmallIntegerField(null=True)),
+                ("date_added", models.DateTimeField(default=django.utils.timezone.now)),
+                (
+                    "alert_rule",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE,
+                        related_name="activation_conditions",
```

Comment:
Is this defined in another pr or created here? (not sure about how these migrations work still 😅

Just curious cause this is what my table looks like:
```
                                      Table "public.sentry_alertrule"
        Column        |           Type           | Collation | Nullable |             Default
----------------------+--------------------------+-----------+----------+----------------------------------
 id                   | bigint                   |           | not null | generated by default as identity
 name                 | text                     |           | not null |
 status               | smallint                 |           | not null |
 include_all_projects | boolean                  |           | not null |
 threshold_period     | integer                  |           | not null |
 date_modified        | timestamp with time zone |           | not null |
 date_added           | timestamp with time zone |           | not null |
 organization_id      | bigint                   |           |          |
 snuba_query_id       | bigint                   |           |          |
 resolve_threshold    | double precision         |           |          |
 threshold_type       | smallint                 |           |          |
 owner_id             | bigint                   |           |          |
 comparison_delta     | integer                  |           |          |
 team_id              | bigint                   |           |          |
 user_id              | bigint                   |           |          |
 monitor_type         | integer                  |           | not null | 0
```

Reply:
okay cool - that's kinda what i thought it was meant to do (create the foreign key only).

I'll try pulling and making sure my db is up to date.

Reply:
This new table has a foreign key to `AlertRule`, not the other way around, so you wouldn't expect to see a change in `AlertRule`

Reply:
@wedamija Yeah - i was trying to find the column in AlertRule that this is a foreign key to - the syntax was making me think that `AlertRule` should have the column `activation_conditions`.